spec/unit/data_providers/hiera_interpolation_spec.rb

```diff
@@ -217,6 +218,11 @@ def expect_lookup(*keys)
       expect_lookup('a.d')
       expect(interpolator.interpolate("a dot f: %{lookup(\"'a.d'\")}", lookup_invocation, true)).to eq('a dot f: ')
     end
+
+    it 'should not find a subkey that is matched within a string' do
+      expect_lookup('key')
```

Comment:
I prefer single quoted strings because they cause less overhead for the Ruby parser/evaluator.